Restructure for Typst Universe

mielpeeters opened the pull request:
I've updated the directory structure to comply with Typst's guidelines for templates and packages.
This way, we should be able to push this to the official Typst Universe!

Also, I updated some details such that the template is now compatible with the `v0.11.0` compiler.

mielpeeters committed:
feat: follow typst package/template format

mielpeeters committed:
fix: remove unnecessary `.nix` and `justfile`

mielpeeters committed:
feat: update some functions and toml declaration

mielpeeters committed:
feat: move README and LICENSE and fill in LICENSE placeholders

CedricMeu commented on 0.1.0/lib.typ:
```diff
@@ -0,0 +1,159 @@
+#let _authors = state("authors", ())
+#let _title = state("title", [])
+#let _supervisors = state("supervisors", ())
+#let _year = state("year", [2023-2024])
```
Might be nice to have the default year be [the current year](https://typst.app/docs/reference/foundations/datetime/)?

CedricMeu commented on 0.1.0/lib.typ:
```diff
@@ -0,0 +1,159 @@
+#let _supervisors = state("supervisors", ())
+#let _year = state("year", [2023-2024])
+
+#let thesis(
```
Don't know if it's useful to have default values for the parameters here. These should all be overridden anyway (except for year, look at previous comment).

mielpeeters replied:
You're right, but to keep these arguments named instead of positional, we should maybe have `none` as default value.
Using this function without named arguments would become messy, I think.

CedricMeu commented on 0.1.0/template/images/ea.png:
I think we can remove the front-page (and therefore these images) for now. It messes up the page numbering.

CedricMeu replied:
Unless you got that working?

CedricMeu commented on 0.1.0/template/main.typ:
```diff
@@ -0,0 +1,29 @@
+#import "@local/ugent-thesis:0.1.0": thesis, acronyms, front-page, page-content
+
+#show: thesis.with(
+  title: [A UGent Master's Dissertation Created Using Typst],
```
Replace "Dissertation" with "Thesis" for consistency?

CedricMeu commented on 0.1.0/lib.typ:
```diff
@@ -0,0 +1,159 @@
+        align(right, emph(body))
+      }
+    },
+    numbering: "I"
```
The numbering is set to roman numerals here, but never set back to normal. Or am i missing something?

CedricMeu commented on 0.1.0/lib.typ:
```diff
@@ -0,0 +1,159 @@
+  }
+}
+
+#let page-content(body) = {
```
Should also reset page numbering here?

CedricMeu commented on 0.1.0/README.md:
```diff
@@ -0,0 +1,26 @@
+## Other remarks
+
+The `front-page` function is really only a placeholder for the eventual front page which you will have to export
+from Plato yourself.
```
Maybe just leave the front page out, might lead to someone not noticing it isn't the correct front page. Would be a painful mistake to make.

CedricMeu reviewed (changes requested):
Thank you very much for this PR!

I have only a few small remarks but these should be easy to fix.

mielpeeters committed:
feat: automatic academic year

mielpeeters committed:
fix: no default argument values

mielpeeters committed:
fix: no front page and reset page numbering

mielpeeters committed:
fix: get rid of bibiliography argument

Users should handle the placement of the bibliography themselves, probably.

mielpeeters committed:
fix: see last

mielpeeters committed:
docs: update readme

remove frontpage remark

mielpeeters commented:
All comments were valid. I solved all issues which you addressed.

CedricMeu commented on 0.1.0/test.typ:
```diff
@@ -0,0 +1,32 @@
+#import "lib.typ": thesis, acronyms, front-page, page-content, show-acronyms
```
`front-page` doesn't exist anymore

CedricMeu replied:
The comments on this file also need to be fixed on `template/main.typ`

CedricMeu commented on 0.1.0/test.typ:
```diff
@@ -0,0 +1,32 @@
+  bibliography: bibliography("template/bib.bib")
+)
+
+#front-page(
```
Same here

CedricMeu commented on 0.1.0/test.typ:
```diff
@@ -0,0 +1,32 @@
+  authors: ("John Doe",),
+  supervisors: ("Prof. Dr. Jane Doe",),
+  year: "2023-2024",
+  bibliography: bibliography("template/bib.bib")
```
Bibliography also isn't an argument anymore

CedricMeu commented on 0.1.0/lib.typ:
```diff
@@ -0,0 +1,125 @@
+        align(right, emph(body))
+      }
+    },
+    numbering: "I"
```
For some reason the roman numbering on the first pages doesn't show op on my build of `test.typ`

CedricMeu replied:
The page numbering (in roman) will start to show up starting on the 3rd page, as if it was the first page.

CedricMeu reviewed (changes requested):
Some of the functions you removed are still in `test.typ` and `main.typ`.  And a problem with numbering.

CedricMeu committed:
fix: removed acrostiche dependency

mielpeeters committed:
Revert "fix: removed acrostiche dependency"

This reverts commit 164dd1672bf4fd7d767be21a7ca300c788ddc857.

mielpeeters committed:
fix: fix numbering and add `todo` feature

note: i reverted the acrostiche removal since it might be better to keep
that as a different pull request

mielpeeters commented:
I fixed the numbering problem!
I also reverted the acrostiche dependency removal because it should maybe be part of a next pull request

CedricMeu commented on 0.1.0/README.md:
```diff
@@ -0,0 +1,25 @@
+
+## Features
+- styling of headings and page numbering, etc.
+- `#todo[your todo note]`: a to-do note that is displayed as a colored block in the output.
```
Love the todo feature, but maybe put it in a different PR and diff against this one.

mielpeeters replied:
✅

CedricMeu commented on 0.1.0/lib.typ:
```diff
@@ -0,0 +1,183 @@
+  
+  // make new sections appear on the right hand side
+  if odd_pagebreaks {
+    set pagebreak(weak: true, to: "odd")
```
how about we add an argument `pagebreak-to`, have it default `"odd"` and pass that to `to`. Or is that too much indirection?

mielpeeters replied:
Will anyone ever set it to something else than "odd" or not setting it at all?
If there's just these two scenarios, a bool is more clear I think

mielpeeters replied:
I kept it as a boolean for now, feel free to change later

CedricMeu commented on 0.1.0/lib.typ:
```diff
@@ -0,0 +1,183 @@
+#let start-numbering(body) = {
+  set heading(numbering: "1.1", supplement: "Chapter")
+
+  show heading.where(level: 1): it => {
```
why is this so complex?

CedricMeu replied:
I mean, if it works alright but i can't wrap my head around it.

mielpeeters replied:
This is for a custom styling of the first-level headings, with a large number and the underlined section name.
The context block is required for getting the counter at that location, and the grid aligns the number and section name next to each other.

CedricMeu replied:
Alright, perfect!

CedricMeu commented on 0.1.0/lib.typ:
```diff
@@ -0,0 +1,183 @@
+
+#let page-content(body) = {
+  // from here, add a new page before each level one heading
+  show heading.where(
+    level: 1
+  ): it => {
+    pagebreak()
+    it
+  }
```
Maybe move this to `thesis`. I prefer that ToCs also get moved to the next odd page.

CedricMeu replied:
Then maybe just remove this function as it only calls `start-numbering`. And try to find a better name for that one. maybe `thesis-content` or `main-content`.

mielpeeters replied:
good ideas

CedricMeu reviewed (changes requested):
My last remarks. I am sure we can finish this PR very soon.

mielpeeters committed:
feat: rename page-content to main-content and pagebreak all first headings

mielpeeters committed:
fix: remove todo feature from this PR

CedricMeu commented:
Nice job!
